keps/sig-cluster-lifecycle/kubeadm/0023-kubeadm-config.md

```diff
@@ -212,6 +214,18 @@ active (a setting already introduced in `v1beta1`) and an ability to ignore some
 Nevertheless, SIG Cluster Lifecycle should be careful not to add fields that are needed for fairly
 limited use cases.
 
+### Public types extraction
+
+Currently, the kubeadm public API types are housed along with conversion and static defaulting code.
+This brings on unnecessary dependencies to vendoring users of those types. Those dependencies are
+mainly on the kubeadm internal types and packages.
+
+To counter this, the public types have to be extracted in separate packages - one for each supported
+version of the kubeadm configuration. Along with those types, the generated deep copy code needs to
+be moved too along with the schema registration code and documentation.
+The old packages are going to reference the new ones and house the conversion and static defaulting code.
+The chosen location for the extracted APIs is [k/kubeadm](https://github.com/kubernetes/kubeadm).
```

Comment:
My last concern here is that if k/kubeadm will be dedicated to the API, then we can't use it when kubeadm itself will move out of k/k.
@rosti, @neolit123 opinions?

Reply:
It can be used. My idea is to just shift the contents of k/k/cmd/kubeadm to k/kubeadm and maintain structure similar to what is in kubernetes/kubernetes#79661 (which is basically `api`, `app` and `test`).
The rest of the stuff that's in k/kubeadm can be left directly under the root dir for a start (then probably shuffled around).
However, this is out of scope for this KEP, hence not mentioned.

Reply:
ok, then if there is no problem for the staging mechanism in having in k/kubeadm both kubeadm and the kubeadm API, ok for me.